src/mudlet-lua/tests/Regex_spec.lua

```
@@ -0,0 +1,274 @@
+    -- optional groups (?: ... )?
+    it("captures an optional group when present", function()
+        local send = spy.on(_G, "send")
+        local snapshot = {}
+        local pattern = "^Name: (\\w+)(?:\\s+Class: (\\w+))?$"
+
+        local id = tempRegexTrigger(pattern, function()
+            send("match")
+            snapshot = matches
+        end, 1)
+
+        feedTriggers("\Name: Tester Class: druid\n")
```

Luacheck (1.2.0)

[error] invalid escape sequence '\N'

(E011)